### PR TITLE
Add `$pullAll` array update operator

### LINKED ISSUE #827: Support `$pullAll` array update operator
https://www.mongodb.com/docs/manual/reference/operator/update/pullAll/

DoD:
- [x] Nested arrays (arrays of arrays) are not in the scope. 
- [x] Arrays of objects are in the scope
- [x] update user doc (supported operators)

Compatibility tests for both valid results and errors are in the scope:
- [x] pullAll from existent array field
- [x] pullAll from non-existent field
- [x] pullAll from a non-array field
- [x] negative test cases 
- [x] dot notation
- [ ] think what other test cases might be needed

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

Closes #827.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/common/update_array_operators.go [hunk 1]
```diff
@@ -315,3 +315,100 @@ func processAddToSetArrayUpdateExpression(doc, update *types.Document) (bool, er
...
+			return false, lazyerrors.Error(err)
+		}
+
+		// If the path does not exist, create a new array and set it.
+		if !doc.HasByPath(path) {
+			if err = doc.SetByPath(path, types.MakeArray(1)); err != nil {
+				return false, commonerrors.NewWriteErrorMsg(
+					commonerrors.ErrUnsuitableValueType,
+					err.Error(),
+				)
```
Comment (rumyantseva) [inline internal/handlers/common/update_array_operators.go:346]: Can we cover is by test? Maybe something where we try to set a value to a non-suitable field? Like `update: bson.D{{"$pullAll", bson.D{{"v.pampam", bson.A{int32(42)}}}}},` where `v` is a scalar, so it's not possible to set `pampam` field

File: internal/handlers/common/update_array_operators.go [hunk 2]
```diff
@@ -315,3 +315,100 @@ func processAddToSetArrayUpdateExpression(doc, update *types.Document) (bool, er
...
+			}
+		}
+
+		val, err := doc.GetByPath(path)
+		if err != nil {
+			return false, err
```
Comment (rumyantseva) [inline internal/handlers/common/update_array_operators.go:352]: Hmm, after passing `HasByPath`, this is rather a lazy error or a panic (as if everything is fine, this line doesn't seem to be reachable).

**Review by rumyantseva — COMMENTED** [review]
Looks good to me overall, added a couple of tiny comments.

File: internal/handlers/common/update_array_operators.go (hunk 1 above)
Comment (w84thesun) [reply internal/handlers/common/update_array_operators.go:346]: Added.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by chilagrow — APPROVED** [review]
looks great!

**Review by Fashander — APPROVED** [review]

**Pull request merged**